### ISSUE TITLE
Fix API-reference in documentation

### ISSUE THREAD

**Issue by pz-max** [pypsa-meets-earth/pypsa-earth#85]
Our documentation is currently **not** build automatically see:
https://pypsa-meets-africa.readthedocs.io/en/latest/api_reference.html
We should fix that before the prototype is released.

**To reproduce the error message:**
- Install the docs environment `conda env create -f envs/environment.docs.yaml`
- Direct to the doc folder `cd /doc`
- Run `make html` to build the documentation locally.
- With the LiveServer Add-on you can open the created HTML at /doc/_build/html/api_reference.html
You can see that the API is not working due to a couple of reasons.

**Some places that can help solving the issue:**
Readthedocs uses Sphinx. In particular, the _sphinx.ext.autodoc_ creates the automated documentation. It is documented [here](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html#module-sphinx.ext.autodoc).

Examples of other API's (automated documentation):
- [PyPSA](https://pypsa.readthedocs.io/en/latest/api_reference.html)
- [PyPSA-Eur](https://github.com/PyPSA/pypsa-eur/blob/master/doc/preparation/build_renewable_profiles.rst), for high level documentation of how rules work 
- [Snakemake](https://snakemake.readthedocs.io/en/stable/api_reference/internal/snakemake.html#module-snakemake.rules)

**Comment by davide-f** [comment]
Commit fd4add803db80190c24fd34ae7b957b3b58da4c4 fixes the issue on the documentation, which now is generated.
In the future we should create the images that are used in the documentation, as done in PyPSA-Eur

Issues where generated by specifying elements in the list `autodoc_mock_imports` in doc/conf.py file.
I commented all the packages listed in that list and the documentation runs.

Moreover, I improved the API_reference to improve the readability.

Note: on my computer it works, but the documentation online here on github is not update; to be finalized
